controller/collaborators.go

```diff
@@ -84,22 +85,29 @@ func (c *CollaboratorsController) List(ctx *app.ListCollaboratorsContext) error
 				}, "unable to find the identity listed in the space policy")
 				return errors.New("Identity listed in the space policy not found")
 			}
-			appUser := ConvertToAppUser(ctx.RequestData, &identities[0].User, &identities[0])
-			data[i] = appUser.Data
+			resultIdentities[i] = identities[0]
+			resultUsers[i] = identities[0].User
 			return nil
 		})
 		if err != nil {
 			return jsonapi.JSONErrorResponse(ctx, goa.ErrInternal(err.Error()))
 		}
 	}
 
-	response := app.UserList{
-		Links: &app.PagingLinks{},
-		Meta:  &app.UserListMeta{TotalCount: count},
-		Data:  data,
-	}
-	setPagingLinks(response.Links, buildAbsoluteURL(ctx.RequestData), len(page), offset, limit, count)
-	return ctx.OK(&response)
+	return ctx.ConditionalEntities(resultUsers, c.config.GetCacheControlCollaborators, func() error {
+		data := make([]*app.UserData, len(s))
+		for i := range resultUsers {
```

Comment:
I guess the reason for extracting the appUser conversion from the iteration above is to not overload the transaction?
That's OK but you don't have to keep two arrays, one for users and one for identities. You can keep the identities only and then use identities[0].User